plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -34,18 +34,32 @@
 import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
 
 public class StandardFacetInstallDelegate extends AppEngineFacetInstallDelegate {
-  private final static String APPENGINE_WEB_XML = "appengine-web.xml";
+  private static final String APPENGINE_WEB_XML = "appengine-web.xml";
 
   @Override
   public void execute(IProject project,
                       IProjectFacetVersion version,
                       Object config,
                       IProgressMonitor monitor) throws CoreException {
     super.execute(project, version, config, monitor);
+    // If Dynamic Web facet was additionally installed together, it will have created "WebContent/"
+    // as the default webapp directory (unless the user changed the default when installing
+    // the Web facet. We detect if there already existed another webapp directory, and if so,
+    // fix the default webapp setting to point to the existing directory.
+    fixWebContentRootIfNecessary(project);
+
     createConfigFiles(project, monitor);
     installAppEngineRuntimes(project);
   }
 
+  private void fixWebContentRootIfNecessary(IProject project) {
+    if (WebProjectUtil.isDefaultWebContentFolderBogus(project)) {
+      Fix/update webapp setting in ".settings/org.eclipse.wst.common.component" (XML). How? To figure out...
```

Comment:
can we do this by setting the relevant facet property as we do now in the new project wizard? That is, 



    IDataModel webModel = DataModelFactory.createDataModel(new WebFacetInstallDataModelProvider());
    webModel.setBooleanProperty(IJ2EEModuleFacetInstallDataModelProperties.ADD_TO_EAR, false);
    webModel.setBooleanProperty(IJ2EEFacetInstallDataModelProperties.GENERATE_DD, false);
    webModel.setBooleanProperty(IWebFacetInstallDataModelProperties.INSTALL_WEB_LIBRARY, false);
    webModel.setStringProperty(IWebFacetInstallDataModelProperties.CONFIG_FOLDER, "src/main/webapp");
    facetedProject.installProjectFacet(WebFacetUtils.WEB_25, webModel, monitor);